### PR TITLE
fix(components): [upload] add before-upload test

### PR DESCRIPTION
**Pull request by chenxch**
#12029 supplementary test

Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/upload/__tests__/upload.test.tsx [hunk 1]
```diff
@@ -131,6 +131,42 @@ describe('<upload />', () => {
...
+    // Ensure that the modified data in before-upload can be correctly passed into the upload request. (#12029)
+    test('in beforeUpload change data correctly to request', async () => {
+      const keyList: string[] = []
+      const beforeUpload = vi.fn((file) => (data.value.key = file.name))
+      const httpRequest = ref(
+        vi.fn((val) => {
+          keyList.push(val?.data?.key)
+          return Promise.resolve()
+        })
+      )
```
Comment (tolking) [inline packages/components/upload/__tests__/upload.test.tsx:143]: The `ref(vi.fn(...))` have any special significance?

Comment (chenxch) [reply packages/components/upload/__tests__/upload.test.tsx:143]: It is just a copy of the previous case transformation.

Comment (tolking) [reply packages/components/upload/__tests__/upload.test.tsx:143]: IMO, `ref()` is redundant here

Comment (chenxch) [reply packages/components/upload/__tests__/upload.test.tsx:143]: Ok, then I will delete the others later.

Comment (chenxch) [reply packages/components/upload/__tests__/upload.test.tsx]: done.

**Review by tolking — APPROVED** [review]

**Review by btea — APPROVED** [review]

**Pull request merged**